package client;

message Hello {
	required string address = 1;
	required int32 port = 2;
	required int32 session_id = 3;
}

message AllocateClient {
	required int32 amount = 1;
	required string batchhash = 2;
}

message AllocateResponse {
	required bool hasdata = 1;
	required bool isavailable = 2;
}

message Data {
	required string batchhash = 1;
	required int32 datasize = 2;
	required int32 datatype = 3;
	required bytes data = 4;
	required string returnaddress = 5;
	required int32 returnport = 6;
}

message Tasks {
	message Task {
		required int32 tasksize = 1;
		required bytes task = 2;
	}

	required string batchhash = 1;
	repeated Task task = 2;
}

message Result {
	required string batchhash = 1;
	required bytes result = 2;
}
